### ISSUE TITLE
[FREQ] Make microphone permission optional

### ISSUE THREAD

**Issue by tiendq** [Mijick/Camera#54]
Assuming your app only uses camera to take photos then there is no reason to tell users that it needs to access micro phone. I think app should not ask for micro phone if `outputType: .photo` and `.outputTypePickerVisible(false)`, it means micro phone is not used anywhere.

Thank you,

**Comment by mpdifran** [comment]
Yeah would really appreciate tighter controls on privacy prompts.

**Comment by SubvertDev** [comment]
@tiendq @mpdifran if you use camera for photos only, you can try my [optional-microphone-permission](https://github.com/SubvertDev/CameraView/tree/feature/optional-microphone-permission) branch, I've added `requireMicrophoneUsage` Bool property to CameraManager init that wraps all calls to microphone.
Sadly I don't have much free time now to check how it plays along with video mode, so it's not complete, but works for me in photo only mode :)

**Comment by FulcrumOne** [comment]
Hey,

OMG, it's been a while since I last worked on this framework and it looks like there's a lot of stuff to do 😅.

I'm just letting you know that I'm starting work on it today; today I'm going to create a backlog, prioritise tasks and start the first sprint, so you'll know which things are going to be fixed / implemented first.

I'm sorry I haven't maintained it for so long, but I had to spend too much time on [Popups](https://github.com/Mijick/Popups), and I literally had no time for this project. However, that will change now 😉

Have a nice day,
Tomasz

**Comment by tiendq** [comment]
> @tiendq @mpdifran if you use camera for photos only, you can try my [optional-microphone-permission](https://github.com/SubvertDev/CameraView/tree/feature/optional-microphone-permission) branch, I've added `requireMicrophoneUsage` Bool property to CameraManager init that wraps all calls to microphone. Sadly I don't have much free time now to check how it plays along with video mode, so it's not complete, but works for me in photo only mode :)

Thank you, I'll try it.

**Comment by tiendq** [comment]
@FulcrumOne I'm just curious about this one, I see it was marked as `In review` 5 days ago but I don't see any related PR, where is your PR?

**Comment by tiendq** [comment]
> @tiendq, I have not yet created a PR as this is part of a larger update I am still working on, however you can use the `patch-3.0.0` branch

Sorry but I tested the latest release `v3.0.0` and it still asked for microphone permission.

```
MCamera()
        .setCameraOutputType(.photo)
        .setAudioAvailability(false)
```

**Comment by tiendq** [comment]
More, even if my app does not need microphone it still must includes `NSMicrophoneUsageDescription` key, right? Because when I removed this key then there was an exception right before app was going to ask for required permissions.

**Comment by FulcrumOne** [comment]
my fault, I haven't noticed that AVDeviceCapture.get asks for permissions on its own. This has been fixed in the `patch-3.0.1` branch.
